homeassistant/components/sms/sensor.py

```diff
@@ -0,0 +1,76 @@
+"""Support for SMS dongle sensor."""
+import logging
+
+import gammu  # pylint: disable=import-error, no-member
+
+from homeassistant.const import DEVICE_CLASS_SIGNAL_STRENGTH
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN, SMS_GATEWAY
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the GSM Signal Sensor sensor."""
+    gateway = hass.data[DOMAIN][SMS_GATEWAY]
+    entities = []
+    imei = await gateway.get_imei_async()
+    name = f"gsm_signal_imei_{imei}"
+    entities.append(GSMSignalSensor(hass, gateway, name,))
+    async_add_entities(entities, True)
+
+
+class GSMSignalSensor(Entity):
+    """Implementation of a GSM Signal sensor."""
+
+    def __init__(
+        self, hass, gateway, name,
```

Comment:
Don't pass in `hass`. It will be set on the entity when it has been added to home assistant.